### DISCUSSION TITLE
Different logs output with sigs.k8s.io/controller-runtime v0.12.1

### DISCUSSION THREAD

**Discussion by hatrena** [kubernetes-sigs/kubebuilder#2769] [Support] [answered]
Currently, with package `sigs.k8s.io/controller-runtime v0.11.0` when I run the manager I can see the following dummy logs:

`1.656069911530586e+09	INFO	controller.GUESTBOOK_CONTROLLER	Starting Controller	{"reconciler group": "webapp.test.example.com", "reconciler kind": "Guestbook"}`

`1.6560699115306501e+09	INFO	controller.FOO_CONTROLLER	Starting EventSource	{"reconciler group": "webapp.test.example.com", "reconciler kind": "Foo", "source": "kind source: *v1.Foo"}`

`1.656069911530703e+09	INFO	controller.FOO_CONTROLLER	Starting Controller	{"reconciler group": "webapp.test.example.com", "reconciler kind": "Foo"}`

`1.656069911631671e+09	INFO	controller.GUESTBOOK_CONTROLLER	Starting workers	{"reconciler group": "webapp.test.example.com", "reconciler kind": "Guestbook", "worker count": 1}`

`1.6560699116320908e+09	INFO	controller.FOO_CONTROLLER	Starting workers	{"reconciler group": "webapp.test.example.com", "reconciler kind": "Foo", "worker count": 1}`

As you can see each line is corresponding with a specific controller and kind (Guestbook and Foo). so far so good.

***

Now if I upgrade the `sigs.k8s.io/controller-runtime` package to `v0.12.1`. All the logs look the same for all controllers and kinds:

`1.65607012707837e+09	INFO	Starting EventSource	{"controller": "FOO_CONTROLLER", "controllerGroup": "webapp.test.example.com", "controllerKind": "Foo", "source": "kind source: *v1.Guestbook"}`

`1.65607012707843e+09	INFO	Starting Controller	{"controller": "FOO_CONTROLLER", "controllerGroup": "webapp.test.example.com", "controllerKind": "Foo"}`

`1.6560701270784159e+09	INFO	Starting EventSource	{"controller": "FOO_CONTROLLER", "controllerGroup": "webapp.test.example.com", "controllerKind": "Foo", "source": "kind source: *v1.Foo"}`

`1.6560701270785e+09	INFO	Starting Controller	{"controller": "FOO_CONTROLLER", "controllerGroup": "webapp.test.example.com", "controllerKind": "Foo"}`

`1.656070127179373e+09	INFO	Starting workers	{"controller": "FOO_CONTROLLER", "controllerGroup": "webapp.test.example.com", "controllerKind": "Foo", "worker count": 1}`

`1.656070127179872e+09	INFO	Starting workers	{"controller": "FOO_CONTROLLER", "controllerGroup": "webapp.test.example.com", "controllerKind": "Foo", "worker count": 1}`

It basically takes info from the last imported controller and uses them for all controllers logs.

Is it supposed to be like this because Kubebuilder is not supporting package `sigs.k8s.io/controller-runtime v0.12.1`or i am missing something?

Best

**Comment by camilamacedo86** [comment]
If you upgrade the go mod on your project and then, build a new manager (Operator) with. The new Operator image will be using the new dep added. 

Kubebuilder is only a CLI tool to help out you scaffold the projects. Then, if you are looking to see differences in the logs because of https://github.com/kubernetes-sigs/controller-runtime/pull/1827/files you need to check if you do not need to change the default scaffold in the main. go to use it.

**Comment by hatrena** [comment, marked as answer]
Working with `sigs.k8s.io/controller-runtime v0.12.1` is still resulting in the odd logs. Apparently, ths issue is resolved with `sigs.k8s.io/controller-runtime v0.12.2`.

Thank you

**Reply by hatrena** [reply]
And i assume this is the related commit that solved the issue: https://github.com/kubernetes-sigs/controller-runtime/commit/19f8d99ed638923efbd1acb92cbd0cb60185cac5

**Comment by camilamacedo86** [comment]
Closing since it is answered.
